azure/packages/test/scenario-runner/src/DocCreatorRunner.ts

```diff
@@ -4,30 +4,56 @@
  */
 import child_process from "child_process";
 
+import { ConnectionState } from "fluid-framework";
+import { AzureClient } from "@fluidframework/azure-client";
 import { TypedEventEmitter } from "@fluidframework/common-utils";
+import { PerformanceEvent } from "@fluidframework/telemetry-utils";
+import { IFluidContainer } from "@fluidframework/fluid-static";
+import { timeoutPromise } from "@fluidframework/test-utils";
 
-import { IRunConfig, IRunner, IRunnerEvents, IRunnerStatus, RunnnerStatus } from "./interface";
-import { delay } from "./utils";
+import {
+	AzureClientConnectionConfig,
+	ContainerFactorySchema,
+	IRunConfig,
+	IRunner,
+	IRunnerEvents,
+	IRunnerStatus,
+	RunnnerStatus,
+} from "./interface";
+import {
+	AzureClientConfig,
+	createAzureClient,
+	delay,
+	getScenarioRunnerTelemetryEventMap,
+	loadInitialObjSchema,
+} from "./utils";
+import { getLogger } from "./logger";
 
-export interface AzureClientConfig {
-	type: "remote" | "local";
-	endpoint?: string;
-	key?: string;
-	tenantId?: string;
-	useSecureTokenProvider?: boolean;
-	region?: string;
-}
-
-export interface DocSchema {
-	initialObjects: { [key: string]: string };
-	dynamicObjects?: { [key: string]: string };
-}
+const eventMap = getScenarioRunnerTelemetryEventMap("DocCreator");
 
 export interface DocCreatorRunnerConfig {
-	connectionConfig: AzureClientConfig;
-	schema: DocSchema;
+	connectionConfig: AzureClientConnectionConfig;
+	schema: ContainerFactorySchema;
 	numDocs: number;
 	clientStartDelayMs: number;
+	client?: AzureClient;
+}
+
+export interface DocCreatorRunnerRunConfig
+	extends IRunConfig,
+		Pick<
+			AzureClientConfig,
+			| "connType"
+			| "connEndpoint"
+			| "tenantId"
```

Comment:
All this endpoint + auth stuff we should be able to source from env and avoid all this parsing repetition. Again, doesn't need to be part of this PR :) We should clean it up at some point though.